src/dali/plugin/pair_converter/csv/kraken.py

```diff
@@ -222,7 +231,10 @@ def _retrieve_cached_bar(self, base_asset: str, quote_asset: str, timestamp: int
                             close=RP2Decimal(row[self.__CLOSE]),
                             volume=RP2Decimal(row[self.__VOLUME]),
                         )
-
+            except FileNotFoundError:
+                self.__logger.error(f"No such file={file_path} (skipping) {timestamp}. Please open an issue at %s %s",
+                                    self.ISSUES_URL, datetime.fromtimestamp(timestamp))
+                should_log = True
```

Comment:
Could you explain why you have this? Anything you don't want to show up and regular logs you can just log as `debug`.

Reply:
I added it because I wanted to understand the exception that occurred and if it would resolve, I think I realize now though that if all time periods fail, it will fall back on the REST api for pairing. I can remove, let me know though. 

Question: Does the Kraken REST API allow queries for price before the last quarter?

Reply:
> I added it because I wanted to understand the exception that occurred and if it would resolve, I think I realize now though that if all time periods fail, it will fall back on the REST api for pairing. I can remove, let me know though.

Anything you used for testing/troubleshooting needs to be removed from production code.

> Question: Does the Kraken REST API allow queries for price before the last quarter?

Yes, it should fallback on the REST API if it doesn't find it in the CSV. That's what __should__ be happening. I suspect that might not be the case if you were having problems with XBT, though. I'll have to look at it when I start work on the aliases.

Reply:
It looks like it was EOSUSD not XBT. My mistake for the confusion. I forgot to put it back in when I was testing without it so its been removed.

Reply:
I'll look at this when I look at aliases. There is a bug here somewhere that I'll try to hunt down.